enhancements/installer/bootstrap-ignition-shim-asset.md

```diff
@@ -0,0 +1,123 @@
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+There are a few platforms that use an Ignition shim for their bootstrap machine. The shim
+is essentially a pointer to the main config that is stored on a remote server, which helps
+us avoid problems with instance metadata size limits on our platform. Previously, we were doing
+this in Terraform using the terraform Ignition Provider, however the repo has fallen extremely out
+of date, making us unable to add additional CA Bundles to our nodes. This means that clients that
+use self signed certificates are unable to use our IPI installer, and can not upgrade to openshift 4 on Openstack.
+We considered both updating the
+terraform provider and creating a new installer asset as possible solutions, and ultimately decided
+that the best solution for our team is to create the installer asset. We plan to create an ignition
+asset similar to the [node asset](https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/node.go)
+for bootstrap ignition shims that other platforms can **opt in** to.
+
+## Motivation
+
+We spent a lot of time looking at the Terraform Ignition Provider to determine if we could
+get involved in the community there to solve our problem. However we had a number of hangups
+with this. First and foremost, that repo has not seen code merged since May 2019. We also noticed
+a number of valid and mergable pull requests waiting for approval for months.
+It seems as if there aren't any maintainers, and we were also unable to reach the contributers.
+The CA Bundle bug is preventing a considerable portion of our IPI users
+from being able to upgrade to OpenShift 4 on OpenStack, and we need this fix in 4.3, so this was rather
+dissuading. Lastly, we discovered that for our UPI solution, we wont have a way to generate the bootstrap
+ignition shim if we continue generating it in terraform. We would like to generate the ignition configs for
+UPI with the command `openshift-install create ignition-configs` so that Ignition configs for both UPI and IPI
+are created the same way.
+
+### Goals
+
+Our goal is to create an installer asset, written in Go, that creates an Ignition
+shim for the bootstrap node. This shim should contain only the data
+required to succesfully fetch the bootstrap Ignition config from its endpoint.
+All further customizations should be added to the bootstrap Ignition config, not
+the bootstrap shim. The design of this asset should be similar to the node ignition
+asset so that other platforms can opt into using it, and add their own customizations.
+
+## Proposal
+
+We are proposing to create a new installer asset in `installer/pkg/asset/ignition/bootstrap/shim.go`.
+This assest would have similar design to the [node asset] (https://github.com/openshift/installer/blob/master/pkg/asset/ignition/machine/node.go),
+and will serve as a pointer to the main bootstrap ignition config. Platforms can further customize this
```

Comment:
so the plan is to create a bootstrap-shim.ign file that has empty endpoint in Go, that the terraform code can edit to fill the actual endpoint?? like for AWS an s3 bucket's file URL created in terraform, Azure/GCP the signed URL to the storage account blob file again created in terraform.

For platforms like vSphere/baremetal, what is the expectation when the user sees this file, modify the ignition to add the expected endpoint?

Reply:
take a look at this proof of concept: https://github.com/openshift/installer/pull/2132

Reply:
@iamemilio If you can point me to the changeset that would answer my question, that would be really nice

Reply:
Also include the example of the shim.ign file here, that definitely useful here.